CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -11,4 +11,155 @@ def get_ddb_connection(env:str=os.getenv('Environment', '')) -> 'botocore.client
+        'neighborhood': 'S',
+        'organizer_email': 'S',
+        'tags': 'L',
+        'food_accepts': 'L',
+        'food_restrictions': 'L',
+        'lat': 'S',
+        'long': 'S',
+        'last_editted': 'N',
+        'profile_image': 'S',
+        'check_in_time': 'S',
+        'check_in_notes': 'S',
+        'check_in_status': 'S',
+        'check_in_image': 'S'
+    }
+
+    def __init__(self, db_client, fridge: dict={}):
+        self.display_name:str = fridge.get('display_name', None) #display_name must be alphanumeric (spaces are fine)
+        self.username:str = fridge.get('username', None)
+        self.fridge_state:str = fridge.get('fridge_state', None)
+        self.address:str = fridge.get('address', None)
+        self.instagram:str = fridge.get('instagram', None)
+        self.info:str = fridge.get('info', None)
+        self.url:str = fridge.get('url', None)
+        self.neighborhood:str = fridge.get('neighborhood', None)
+        self.organizer_email:str = fridge.get('organizer_email', None)
+        self.tags:list = fridge.get('tags', None)
+        self.food_accepts:list = fridge.get('food_accepts', None)
+        self.food_restrictions:list = fridge.get('food_restrictions', None)
+        self.lat:str = fridge.get('lat', None)
+        self.long:str = fridge.get('long', None)
+        self.profile_image:str = fridge.get('profile_image', None)
+        self.last_editted:int = fridge.get('last_editted', None)
+        self.check_in_time:int = fridge.get('check_in_time', None)
+        self.check_in_notes:str = fridge.get('check_in_notes', None)
+        self.check_in_status:str = fridge.get('check_in_status', None)
+        self.check_in_image:str = fridge.get('check_in_image', None)
+        self.db_client = db_client
+        if self.fridge_state is not None:
+            self.fridge_state = self.fridge_state.upper()
+
+    def has_required_fields(self) -> bool:
+        for field in self.REQUIRED_FIELDS:
+            if getattr(self, field) is None:
+                raise RequiredFieldMissingException(object_name='fridge', field=field)
+        return True
+
+    def is_valid_fridge_state(self) -> bool:
+        if self.fridge_state not in self.STATES:
+            raise InvalidStateException(fridge_state=self.fridge_state)
+        return True
+
+    def get_fridge(db_client):
+        pass
+
+    def get_fridges(db_client):
+        pass
+
+    def add_fridges(db_client):
+        pass
+    
+    def set_username(self):
+        username = self.display_name.lower().replace(" ", "")
+        if not username.isalnum():
+            raise InvalidDisplayNameException(username)
+        self.username = username
+
+
+    def build_fridge_item(self):
+        fridge_item = {}
+        for key in self.FRIDGE_TYPES:
+            val = getattr(self, key)
+            if val is not None:
+                fridge_item[key] = {self.FRIDGE_TYPES[key]: val}
+        return fridge_item
+
+
+    def add_fridge(self):
```

Comment:
could rename add_fridge() and get_fridge() to add() and get() and put that in a base class since all of the database classes will have a similar function?

Reply:
Yes, good idea. `add/get` sounds very short though, how about something like `add_object` or `add_record`?

Reply:
named it `add_item` and `get_item` to follow the naming convention that dynamodb uses